src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -2640,9 +2640,9 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                                             ASR::ArrayConstant_t *a = ASR::down_cast<ASR::ArrayConstant_t>(cast->m_arg);
                                             ASR::ttype_t* real_type = cast->m_type;
                                             Vec<ASR::expr_t*> body;
-                                            body.reserve(al, a->n_args);
-                                            for (size_t i = 0; i < a->n_args; i++) {
-                                                ASR::expr_t *e = a->m_args[i];
+                                            body.reserve(al, ASRUtils::get_fixed_size_of_array(a->m_type));
+                                            for (size_t i = 0; i < (size_t) ASRUtils::get_fixed_size_of_array(a->m_type); i++) {
```

Comment:
```suggestion
                                            body.reserve(al, a->m_n_data / 8);
                                            for (size_t i = 0; i < a->m_n_data / 8; i++) {
```

This can be replaced like this, I'll prefer to do code clean up in subsequent PRs once CI works.